filters/fadein/fadein.go

```diff
@@ -0,0 +1,252 @@
+package fadein
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/routing"
+)
+
+const (
+	DurationName        = "fadeInDuration"
+	EaseName            = "fadeInEase"
+	EndpointCreatedName = "endpointCreated"
+)
+
+type (
+	duration time.Duration
+	ease     float64
+
+	endpointCreated struct {
+		when  time.Time
+		which string
+	}
+
+	detectedFadeIn struct {
+		when       time.Time
+		duration   time.Duration
+		lastActive time.Time
+	}
+
+	postProcessor struct {
+		detected map[string]detectedFadeIn
+	}
+)
+
+func NewDuration() filters.Spec {
+	return duration(0)
+}
+
+func (duration) Name() string { return DurationName }
+
+func (duration) CreateFilter(args []interface{}) (filters.Filter, error) {
+	if len(args) != 1 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	switch v := args[0].(type) {
+	case int:
+		return duration(v * int(time.Millisecond)), nil
+	case float64:
+		return duration(int(v) * int(time.Millisecond)), nil
+	case string:
+		d, err := time.ParseDuration(v)
```

Comment:
I wonder why we need the other cases and can't just always use string

Reply:
It supports programmatic definition of routes when you use skipper as a library and want to skip the parsing. I think about it as a nicety for library usage, but also useful in the test code.

Reply:
Ok thanks!